project/src/main/java/com/google/sps/Servlets/SharingServlet.java

```diff
@@ -0,0 +1,35 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.*;
+import com.google.sps.models.Database;
+import com.google.sps.models.User;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+@WebServlet("/Share")
+@MultipartConfig
+public class SharingServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String email = request.getParameter("email");
+    String hash = request.getParameter("documentHash");
+    PrintWriter out = response.getWriter();
+
+    System.out.println(email + hash);
+
+    if(Database.shareDocument(hash, email)) {
```

Comment:
for readability it might be better to save Database.shareDocument() into a boolean, eg 
boolean shareSuccessful = Database.shareDocument()...

Reply:
@frankpzn or @HidyHan which is better for readability? I could go either way

Reply:
No strong preference. The condition is pretty straightforward here.

Reply:
I think it's better to extract it as bool, as we generally discourage calling the same function twice.